catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/provider/SolrProviderTestBase.java

```diff
@@ -0,0 +1,195 @@
+    QueryRequest request = new QueryRequestImpl(query);
+    SourceResponse response = provider.query(request);
+
+    assertEquals(count, response.getResults().size());
+  }
+
+  protected DeleteResponse delete(String identifier) throws IngestException {
+    return delete(new String[] {identifier});
+  }
+
+  protected DeleteResponse delete(String[] identifier) throws IngestException {
+    return provider.delete(new DeleteRequestImpl(identifier));
+  }
+
+  protected UpdateResponse update(String id, Metacard metacard) throws IngestException {
+    String[] ids = {id};
+    return update(ids, Collections.singletonList(metacard));
+  }
+
+  protected UpdateResponse update(String[] ids, List<Metacard> list) throws IngestException {
+    return provider.update(new UpdateRequestImpl(ids, list));
+  }
+
+  protected CreateResponse create(Metacard metacard) throws IngestException {
+    return create(Collections.singletonList(metacard));
+  }
+
+  protected CreateResponse create(List<Metacard> metacards) throws IngestException {
+    return createIn(metacards, provider);
+  }
+
+  void addMetacardWithModifiedDate(DateTime now) throws IngestException {
+    List<Metacard> list = new ArrayList<>();
+    MockMetacard m = new MockMetacard(Library.getFlagstaffRecord());
+    m.setEffectiveDate(dateNow(now));
+    list.add(m);
+    create(list);
+  }
+
+  List<Result> getResultsForFilteredQuery(Filter filter) throws UnsupportedQueryException {
+    QueryImpl query = new QueryImpl(filter);
+
+    SourceResponse sourceResponse = provider.query(new QueryRequestImpl(query));
+    return sourceResponse.getResults();
+  }
+
+  Date dateAfterNow(DateTime now) {
+    return now.plusSeconds(A_LITTLE_WHILE).toDate();
+  }
+
+  Date dateBeforeNow(DateTime now) {
+    return now.minusSeconds(A_LITTLE_WHILE).toDate();
+  }
+
+  Date dateNow(DateTime now) {
+    return now.toDate();
+  }
+
+  Set<AttributeDescriptor> numericalDescriptors(
+      String doubleField, String floatField, String intField, String longField, String shortField) {
+    Set<AttributeDescriptor> descriptors = new HashSet<>();
+    descriptors.add(
+        new AttributeDescriptorImpl(Metacard.ID, true, true, true, false, BasicTypes.STRING_TYPE));
+    descriptors.add(
+        new AttributeDescriptorImpl(doubleField, true, true, true, false, BasicTypes.DOUBLE_TYPE));
+    descriptors.add(
+        new AttributeDescriptorImpl(floatField, true, true, true, false, BasicTypes.FLOAT_TYPE));
+    descriptors.add(
+        new AttributeDescriptorImpl(intField, true, true, true, false, BasicTypes.INTEGER_TYPE));
+    descriptors.add(
+        new AttributeDescriptorImpl(longField, true, true, true, false, BasicTypes.LONG_TYPE));
+    descriptors.add(
+        new AttributeDescriptorImpl(shortField, true, true, true, false, BasicTypes.SHORT_TYPE));
+    return descriptors;
+  }
+
+  private static CreateResponse createIn(List<Metacard> metacards, SolrCatalogProvider solrProvider)
```

Comment:
❓ Any reason we're taking in a `SolrCatalogProvider` instead of using the existing `provider` member variable like we do in the other methods?